include/uv.h

```diff
@@ -1664,6 +1677,30 @@ union uv_any_req {
 };
 #undef XX
 
+#if defined(DISCORD_ENABLE_NETMAP)
+typedef struct nm_desc nm_desc_t;
+typedef struct uv_netmap_s uv_netmap_t;
+#define NM_NUM_UDP_SOCKETS (1 << 16)
```

Comment:
Is such a large number here intended? `uv_netmap_s` is a very heavy `struct`.

Reply:
I just went with the total possible number of port numbers for the initial draft. Would it be any better if it were a pointer in the struct rather than the array itself? For generality purposes it's probably easiest to allow any port number or else we might want to think about some kind of mapping container, but it is C :(

Reply:
It's fine -- this is basically alloc'd and left alone forever -- I was just calling it out because the count made me do a double take. We're not hurting for RAM on our worker boxes after-all, right?

Reply:
Yeah in the grand scheme of things, this is 64k * (sizeof(ptr) + sizeof(int)) which is something like 384kB of memory per worker. When I've tested this, total mem util for all workers combined was less than 2% of host memory :D